lib/rubygems/local_remote_options.rb

```diff
@@ -24,8 +24,10 @@ def accept_uri_http
         raise OptionParser::InvalidArgument, value
       end
 
-      unless ['http', 'https', 'file', 's3'].include?(uri.scheme)
-         raise OptionParser::InvalidArgument, value
+      allowed_schemes = ['http', 'https', 'file', 's3']
+      unless allowed_schemes.include?(uri.scheme)
+        msg =  "#{value}\nPreface URLs with one of #{allowed_schemes}"
+        raise OptionParser::InvalidArgument, msg
```

Comment:
It looks like even if I try to create my own custom ```OptionParser::InvalidArgument``` object it still returns whatever is in the ```value``` iterator.

```ruby
err = OptionParser::InvalidArgument.new('custom message')
raise err
```

The above attempt still results in the error message just being whatever ```value``` was when the exception was raised.